src/jobs/terraform_module_lint.yml

```diff
@@ -0,0 +1,60 @@
+description: |
+  terraform linting
+
+executor: terraform
+
+parameters:
+  lint-dir:
+    type: string
+    default: .
+    description: |
+      Directory within to lint all terraform files,
+      defaults to the working directory (.)
+
+  tf-workspace:
+    type: string
+    default: dev
+    description: |
+      Terraform workspace to use
+
+steps:
+  - update_alpine_and_checkout
+  # Terraform Cloud Access
+  - run: |
+      echo "# Terraform CLI config" > ~/.terraformrc
+      echo "credentials \"app.terraform.io\" {" >> ~/.terraformrc
+      echo "  token = \"$TFIO_TOKEN\"" >> ~/.terraformrc
+      echo "}" >> ~/.terraformrc
+  # AWS Cloud Access
+  - run: |
+      mkdir ~/.aws
+      echo "[default]" > ~/.aws/credentials
+      echo "aws_access_key_id = $AWS_ACCESS_KEY_ID-<<parameters.tf-workspace>>" >> ~/.aws/credentials
+      echo "aws_secret_access_key = $AWS_SECRET_ACCESS_KEY-<<parameters.tf-workspace>>" >> ~/.aws/credentials
+  # Azure Cloud Access
+  - run: |
+      mkdir ~/.azure
+      echo "[default]"
+      echo "subscription_id=$AZURE_SUBSCRIPTION_ID-<<parameters.tf-workspace>>]" > ~/.azure/credentials
+      echo "client_id=$AZURE_CLIENT_ID-<<parameters.tf-workspace>>" >> ~/.azure/credentials
+      echo "secret=$AZURE_CLIENT_SECRET-<<parameters.tf-workspace>>" >> ~/.azure/credentials
+      echo "tenant_id=$AZURE_TENANT_ID-<<parameters.tf-workspace>>" >> ~/.azure/credentials
+  # GITHUB Private Repo Access
+  - run: |
+      echo "https://$GITHUB_USER:$GITHUB_TOKEN@github.com" > ~/.git-credentials
+      echo "[user]"  > ~/.gitconfig
+      echo "  name = $GITHUB_USER" >> ~/.gitconfig
+      echo "[credential]" >> ~/.gitconfig
+      echo "  helper = store" >> ~/.gitconfig
+  # Real work
+  - run: terraform workspace new <<parameters.tf-workspace>>
+  - run: terraform init -upgrade <<parameters.lint-dir>>
+  - run: 
+      command: terraform validate <<parameters.lint-dir>>
+      environment: 
+        AWS_DEFAULT_REGION: us-west-2
```

Comment:
I'd add this as an environment variable parameter in the tf_lint job instead of needing to create a new job, or as a context environment variable in circleci.